openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIHeaderMapper.java

```diff
@@ -82,14 +87,23 @@
         List<Parameter> parameters = new ArrayList<>();
         String headerName = headerParam.paramName().get().text().replaceAll("\\\\", "");
         HeaderParameter headerParameter = new HeaderParameter();
-        StringSchema stringSchema = new StringSchema();
-        if (headerParam.expression().kind() == SyntaxKind.STRING_LITERAL) {
-            stringSchema.setDefault(headerParam.expression().toString().replaceAll("\"", ""));
+        Schema<?> headerTypeSchema = ConverterCommonUtils.getOpenApiSchema(
+                headerParam.typeName().toString().replaceAll("\\?", "").
+                        replaceAll("\\[", "").replaceAll("\\]", "").trim());
+        String defaultValue = headerParam.expression().toString().trim().replaceAll("\"", "");
+        List<SyntaxKind> allowedTypes = new ArrayList<>();
+        allowedTypes.addAll(Arrays.asList(SyntaxKind.STRING_LITERAL, SyntaxKind.NUMERIC_LITERAL,
```

Comment:
current http header implementation allows accepting `int` , `boolean` and `string`. since it has been validated through the compilation in the http module. I directly assume this `NUMERIC_LITERAL` is related to int. :)